scenario_player/tasks/blockchain.py

```diff
@@ -171,23 +173,97 @@ def _run(self, *args, **kwargs) -> Dict[str, Any]:  # pylint: disable=unused-arg
         if self.event_args:
             for key, value in self.event_args.items():
                 if "participant" in key:
-                    if isinstance(value, int) or (isinstance(value, str) and value.isnumeric()):
-                        # Replace node index with eth address
-                        self.event_args[key] = self._runner.get_node_address(int(value))
+                    self.event_args[key] = self._get_node_address(value)
 
             event_args_items = self.event_args.items()
             # Filter the events by the given event args.
             # `.items()` produces a set like object which supports intersection (`&`)
             events = [e for e in events if e["args"] and event_args_items & e["args"].items()]
 
-        # Raise exception when events do not match
-        if not self.num_events == len(events):
-            raise ScenarioAssertionError(
-                f"Expected number of events ({self.num_events}) did not match the number "
-                f"of events found ({len(events)})"
+        if self.num_events is not None:
+            # Raise exception when events do not match
+            if not self.num_events == len(events):
+                raise ScenarioAssertionError(
+                    f"Expected number of events ({self.num_events}) did not match the number "
+                    f"of events found ({len(events)})"
+                )
+        return {"events": events}
+
+    def _get_node_address(self, value):
+        if isinstance(value, int) or (isinstance(value, str) and value.isnumeric()):
+            # Replace node index with eth address
+            return self._runner.get_node_address(int(value))
+        return value
+
+
+class AssertChannelSettledEvent(AssertBlockchainEventsTask):
+    _name = "assert_channel_settled_event"
+    SYNCHRONIZATION_TIME_SECONDS = 0
+    DEFAULT_TIMEOUT = 5 * 60  # 5 minutes
+
+    def __init__(
+        self, runner: scenario_runner.ScenarioRunner, config: Any, parent: "Task" = None
+    ) -> None:
+        config.update({"contract_name": "TokenNetwork", "event_name": "ChannelSettled"})
+        super().__init__(runner, config, parent)
+
+        verify_config(config, required_keys=["initiator", "partner", "channel_info_key"])
+
+        self.initiator = self._get_node_address(config["initiator"])
+        self.initiator_amount = config.get("initiator_amount", None)
+        self.partner = self._get_node_address(config["partner"])
+        self.partner_amount = config.get("partner_amount", None)
+
+    def _run(self, *args, **kwargs) -> Dict[str, Any]:  # pylint: disable=unused-argument
+        channel_infos = self._runner.task_storage[STORAGE_KEY_CHANNEL_INFO].get(
+            self._config["channel_info_key"]
+        )
+        if channel_infos is None:
+            raise ScenarioError(
+                f"No stored channel info found for key '{self._config['channel_info_key']}'."
```

Comment:
Might be better to factor out `self._config['channel_info_key']` as a new variable, e.g. `key` and use that in the two places.